### PR TITLE
Support for renaming inputs/outputs on nodegraphs.

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: source/MaterialXRuntime/Private/PvtPrim.h [hunk 1]
```diff
@@ -117,6 +119,8 @@ class PvtPrim : public PvtObject
 
     void removeAttribute(const RtToken& name);
 
+    void renameAttribute(const RtToken& name, const RtToken& newName);
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/Private/PvtPrim.h:122]: General support for rename attributes and relationships in private API.

File: source/MaterialXRuntime/RtNodeGraph.h [hunk 2]
```diff
@@ -31,12 +31,18 @@ class RtNodeGraph : public RtNode
...
+    void renameInput(const RtToken& name, const RtToken& newName);
+
     /// Add an output attribute to the graph.
     RtOutput createOutput(const RtToken& name, const RtToken& type, uint32_t flags = 0);
 
     /// Remove an output attribute from the graph.
     void removeOutput(const RtToken& name);
 
+    /// Rename an output attribute in the graph.
+    void renameOutput(const RtToken& name, const RtToken& newName);
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtNodeGraph.h:44]: In public API only expose rename for inputs and outputs on NodeGraphs for now, which we know we need. 
(Renaming attributes on NodeDefs and Nodes has more complications so we'll wait until/if we need it).

File: source/MaterialXRuntime/Private/PvtPrim.cpp [hunk 3]
```diff
@@ -117,6 +117,21 @@ void PvtPrim::removeRelationship(const RtToken& name)
     }
 }
 
+void PvtPrim::renameRelationship(const RtToken& name, const RtToken& newName)
+{
+    if (getRelationship(newName))
+    {
+        throw ExceptionRuntimeError("A relationship named '" + newName.str() + "' already exists in prim '" + getName().str() + "'");
```
Comment (feldstj) [inline source/MaterialXRuntime/Private/PvtPrim.cpp:124]: Is it the callers responsibility to make sure the name is unique? Or should this API find the next available name?

Comment (niklasharrysson) [reply source/MaterialXRuntime/Private/PvtPrim.cpp:124]: Good point. I could change the behavior to allow name collisions and resolve them internally, as we do when renaming prims in a stage.

**Comment by niklasharrysson** [comment]
Ok, I will do the name resolve in another change list. Will merge this one as soon as CI is completed.

**Pull request merged**